### DISCUSSION TITLE
Craft 4.9 & 5.2 asyncCsrfInputs setting

### DISCUSSION THREAD

**Discussion by gjhead** [solspace/craft-freeform#1424] [Questions & Answers] [answered]
_"Craft 4.9 and 5.2 added the [asyncCsrfInputs](https://craftcms.com/docs/5.x/reference/config/general.html#asynccsrfinputs) config setting, which makes it easy to include CSRF inputs in forms without sacrificing cachability. When that’s enabled, csrfInput() will return a <craft-csrf-input> placeholder element rather than an input, and registers JS code that fetches a CSRF token via Ajax and swaps out the placeholder with an input on page load."_

Does this new setting mean that we could drop the additiona javascript snippet as [shared here](https://docs.solspace.com/craft/freeform/v5/templates/caching/)?

**Comment by jannisborgers** [comment, marked as answer]
Hi @gjhead I just validated that once I set `->asyncCsrfInputs(true)` in `general.php`, I did not have to manually fetch the CSRF token as described in the docs. Instead, Freeform (via Craft’s `csrfInput()`) gets the user session from the server and fills in the CSRF token name and token value automatically. 

This is with Craft 5.4.8 and Freeform 5.6.5. Pull request #1465 removed the manual csrf input that Freeform used before with Craft’s `csrfInput()` function, which reacts to the config setting mentioned above.
